src/System.Management.Automation/engine/hostifaces/HostUtilities.cs

```diff
@@ -522,8 +522,18 @@ internal static string RemoveGuidFromMessage(string message, out bool matchPatte
         internal static string RemoveIdentifierInfoFromMessage(string message, out bool matchPattern)
         {
             matchPattern = false;
-            if (string.IsNullOrEmpty(message))
+            if (message is null ||
+                message.Length < 36 ||
+                message[33] != '[' ||
+                message[32] != ':' ||
+                message[20] != '-' ||
+                message[16] != '-' ||
+                message[12] != '-' ||
+                message[8] != '-')
```

Comment:
```suggestion
                message[8] != '-' ||
                message[12] != '-' ||
                message[16] != '-' ||
                message[20] != '-' ||
                message[32] != ':' ||
                message[33] != '[')
```
It is not necessary to check in reverse order, see [sharplab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8BiAOwFcAbS7YSmAAgBMBLXW+gWACgABAZga8ATAwDCPAN48Gshi1IYYUUtkpCAjEk0AGBgFkAFLw178MXOwDmMAJQy507nJfyAZg0PnL2G/NwMFNQMhIQOrq7e1jAAdAAyMKRWGAAWDAA8DPzaoeERclG+MADa/PwAugwAhAC8DADkxfUhYc75LoU2pcKVtQ0gzblt7bKdJcI6vXX1cIOtIwUW0cVaUw2zLXntYys91dMbQwujS0XFABxrM/X2w65Ox0IA7AxjANxbcgC+n7K8L0FKB87j87golCo1JptCY9PoNMZTK9TjZbi4HvkWB4vCjGGxAlR1EdjmN4olkmlMtlNndtriLpV8aQIBh1nNfpF6Ro9n1rjTHjtVvs2fySfSJldDvMxT4uvwedMBqKFjsylcmhyXGj8hiFv9kbKYMD2qD2vrAcbvjwvkA===).